components/lcd/esp_lcd_ili9881c/esp_lcd_ili9881c.c

```diff
@@ -302,151 +110,436 @@ esp_err_t esp_lcd_new_panel_ili9881c(const esp_lcd_panel_io_handle_t io, const e
+    uint8_t lane_command = ILI9881C_DSI_2_LANE;
+    bool is_command0_enable = false;
+    bool is_cmd_overwritten = false;
+
+    switch (ili9881c->lane_num) {
+    case 0:
+    case 2:
+        lane_command = ILI9881C_DSI_2_LANE;
+        break;
+    case 3:
+    case 4:
+        lane_command = ILI9881C_DSI_3_4_LANE;
+        break;
+    default:
+        ESP_LOGE(TAG, "Invalid lane number %d", ili9881c->lane_num);
+        return ESP_ERR_INVALID_ARG;
+    }
 
-    const ili9881c_lcd_init_cmd_t *init_cmds = vendor_specific_init_code_default;
-    uint16_t init_cmds_size = sizeof(vendor_specific_init_code_default) / sizeof(ili9881c_lcd_init_cmd_t);
+    // back to CMD_Page 1
+    ESP_RETURN_ON_ERROR(esp_lcd_panel_io_tx_param(io, ILI9881C_CMD_CNDBKxSEL, (uint8_t[]) {
+        ILI9881C_CMD_BKxSEL_BYTE0, ILI9881C_CMD_BKxSEL_BYTE1, ILI9881C_CMD_BKxSEL_BYTE2_PAGE1
+    }, 3), TAG, "send command failed");
+    ESP_RETURN_ON_ERROR(esp_lcd_panel_io_tx_param(io, ILI9881C_PAD_CONTROL, (uint8_t[]) {
+        lane_command,
+    }, 1), TAG, "send command failed");
 
     // back to CMD_Page 0
-    ESP_RETURN_ON_ERROR(esp_lcd_panel_io_tx_param(io, 0xFF, (uint8_t[]) {
-        0x98, 0x81, 0x00
+    ESP_RETURN_ON_ERROR(esp_lcd_panel_io_tx_param(io, ILI9881C_CMD_CNDBKxSEL, (uint8_t[]) {
+        ILI9881C_CMD_BKxSEL_BYTE0, ILI9881C_CMD_BKxSEL_BYTE1, ILI9881C_CMD_BKxSEL_BYTE2_PAGE0
     }, 3), TAG, "send command failed");
     // exit sleep mode
     ESP_RETURN_ON_ERROR(esp_lcd_panel_io_tx_param(io, LCD_CMD_SLPOUT, NULL, 0), TAG,
                         "io tx param failed");
     vTaskDelay(pdMS_TO_TICKS(120));
 
-    for (int i = 0; i < init_cmds_size; i++) {
-        ESP_RETURN_ON_ERROR(esp_lcd_panel_io_tx_param(io, init_cmds[i].cmd, init_cmds[i].data, init_cmds[i].data_bytes), TAG, "send command failed");
-    }
-
     ESP_RETURN_ON_ERROR(esp_lcd_panel_io_tx_param(io, LCD_CMD_MADCTL, (uint8_t[]) {
         ili9881c->madctl_val,
     }, 1), TAG, "send command failed");
     ESP_RETURN_ON_ERROR(esp_lcd_panel_io_tx_param(io, LCD_CMD_COLMOD, (uint8_t[]) {
         ili9881c->colmod_val,
     }, 1), TAG, "send command failed");
 
+    // vendor specific initialization, it can be different between manufacturers
+    // should consult the LCD supplier for initialization sequence code
+    if (ili9881c->init_cmds) {
+        init_cmds = ili9881c->init_cmds;
+        init_cmds_size = ili9881c->init_cmds_size;
+    } else {
+        init_cmds = vendor_specific_init_default;
+        init_cmds_size = sizeof(vendor_specific_init_default) / sizeof(ili9881c_lcd_init_cmd_t);
+    }
+
+    for (int i = 0; i < init_cmds_size; i++) {
+        // Check if the command has been used or conflicts with the internal
+        if (is_command0_enable && init_cmds[i].data_bytes > 0) {
+            switch (init_cmds[i].cmd) {
+            case LCD_CMD_MADCTL:
+                is_cmd_overwritten = true;
+                ili9881c->madctl_val = ((uint8_t *)init_cmds[i].data)[0];
```

Comment:
if `ili9881c->madctl_val` already equals to `((uint8_t *)init_cmds[i].data)[0]`, should we still call this a *conflict*?

Reply:
The purpose of this check is that we do not want users to manipulate the MADCTL and COLMOD registers in the initialization parameters; these two registers should be managed by the driver. If they are manipulated during initialization, we want to provide a corresponding warning.